connectors/sources/sharepoint_online.py

```diff
@@ -299,7 +299,7 @@ async def wrapped(*args, **kwargs):
                     async for item in func(*args, **kwargs, retry_count=retry):
                         yield item
                     break
-                except (NotFound, BadRequestError):
+                except (NotFound, BadRequestError, PermissionsMissing):
```

Comment:
It seems like adding `PermissionsMissing` is not related to the current PR. Could you please expand on this?

Reply:
Correct, not related. Just feel we shouldn't retry if there's a PermissionsMissing error.